Tests/SonarScanner.MSBuild.Shim.Test/SonarScannerWrapperTests.cs

```diff
@@ -257,28 +267,78 @@ public void SonarScanner_UserSpecifiedEnvVars_OnlySONARSCANNEROPTSIsPassed()
         }
 
         [TestMethod]
-        public void WrapperError_Success_NoStdErr()
-        {
+        public void WrapperError_Success_NoStdErr() =>
             TestWrapperErrorHandling(executeResult: true, addMessageToStdErr: false, expectedOutcome: true);
-        }
 
         [TestMethod]
         [WorkItem(202)] //SONARMSBRU-202
-        public void WrapperError_Success_StdErr()
-        {
+        public void WrapperError_Success_StdErr() =>
             TestWrapperErrorHandling(executeResult: true, addMessageToStdErr: true, expectedOutcome: true);
+
+        [TestMethod]
+        public void WrapperError_Fail_NoStdErr() =>
+            TestWrapperErrorHandling(executeResult: false, addMessageToStdErr: false, expectedOutcome: false);
+
+        [TestMethod]
+        public void WrapperError_Fail_StdErr() =>
+            TestWrapperErrorHandling(executeResult: false, addMessageToStdErr: true, expectedOutcome: false);
+
+        [TestMethod]
+        public void FindScannerExe_FindsCliZip_ExtractsItSuccessfully()
+        {
+            // Arrange
+            var logger = new TestLogger();
+            MockScannerCliZipFolder(TestContext.TestDir, "99.99");
+
+            // Act
+            SonarScannerWrapper.FindScannerExe(logger, TestContext.TestDir, "99.99");
+
+            // Assert
+            Directory.Exists(Path.Combine(TestContext.TestDir, "sonar-scanner-99.99"));
+            logger.AssertInfoMessageExists("Unzipping sonar-scanner-cli-99.99.zip");
         }
 
         [TestMethod]
-        public void WrapperError_Fail_NoStdErr()
+        public void FindScannerExe_ZipIsAlreadyExtracted_DoesNotExtractAgain()
         {
-            TestWrapperErrorHandling(executeResult: false, addMessageToStdErr: false, expectedOutcome: false);
+            // Arrange
+            var logger = new TestLogger();
+            var scannerCliTestDirectory = TestUtils.CreateTestSpecificFolderWithSubPaths(TestContext, new string[] { "sonar-scanner-99.99", "bin" });
+            TestUtils.CreateEmptyFile(scannerCliTestDirectory, "sonar-scanner.bat");
+            var scannerCliTestDirectoryRoot = Directory.GetParent(scannerCliTestDirectory).Parent.FullName;
+            TestUtils.CreateEmptyFile(scannerCliTestDirectoryRoot, "sonar-scanner-cli-99.99.zip");
+
+            // Act
+            SonarScannerWrapper.FindScannerExe(logger, scannerCliTestDirectoryRoot, "99.99");
```

Comment:
can you also check that the result of this function, to make sure it's the expected one?